src/QsCompiler/BondSchemas/Protocols.cs

```diff
@@ -30,6 +30,7 @@ public static class Protocols
         /// Deserializes a Q# compilation object from its Bond simple binary representation.
         /// </summary>
         /// <param name="byteArray">Bond simple binary representation of a Q# compilation object.</param>
+        /// <remarks>This must not be invoked through a <see cref="Task"/>.</remarks>
```

Comment:
Maybe a comment like this would be clearer:

```suggestion
        /// <remarks>This method waits for <see cref="Task"/>s to complete and may deadlock if invoked through a <see cref="Task"/>.</remarks>
```

Reply:
Yes, that is much better! Thanks for the suggestion :) I'll change it in the other comments too.